google/cloud/bigquery/table.py

```diff
@@ -2005,13 +2005,13 @@ class TimePartitioning(object):
     """
 
     def __init__(
-        self, type_=None, field=None, expiration_ms=None, require_partition_filter=None
+        self, type=None, field=None, expiration_ms=None, require_partition_filter=None
```

Comment:
Modifying a signature of a public method can break user code, unfortunately, and I don't know if the gain (evaluability of repr) is worth it. The linter might also complain, because the new parameter name shadows the `type` builtin.

Reply:
Yes, I agree, I will revert this and propose a change to output repr with `type_ ` parameter instead.